### PR TITLE
BENCH-377: Create Payments & Order Summary Process

### PR DESCRIPTION
**Pull request by mbaltrimas**
image
image
image
image

atm its full of placeholder data as checkout hasn't been fully implemented on the backend yet

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <link rel="preconnect" href="https://fonts.googleapis.com" />
     <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
     <link href="https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;600&display=swap" rel="stylesheet" />
+    <link href="https://fonts.googleapis.com/css?family=Poly" rel="stylesheet" />
 
     <title>Vite App</title>
```
Comment (bethcryer) [inline index.html]: not really related to this PR, but thought this title ought to be changed :P

File: src/components/CheckoutTabs/CheckoutConfirmationTab.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+      <CheckoutSection title="Order">
+        <div className="flex w-full flex-col gap-2">
+          <OrderDetailsMinimal items={localOrder.lineItems} />
+        </div>
+      </CheckoutSection>
+
+      <CheckoutSection title="Order details">
+        <div className="flex w-full flex-row justify-between border-b-2 text-[12px]">
+          <span>Joe Bloggs</span>
```
Comment (bethcryer) [inline src/components/CheckoutTabs/CheckoutConfirmationTab.tsx]: it's not necessary, but you could make sure you have all your placeholder data in variables for now, that way when it comes to using real data down the line it'll be harder to accidentally miss places that need changing

**Review by bethcryer — DISMISSED** [review]

File: src/components/CheckoutTabs/CheckoutSummaryTab.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    if (!localOrder.id) {
+      createOrder.mutate(createdOrder, {
+        onSuccess: () => {
+          setCurrentTab(CheckoutTabType.Confirmation);
+        },
+      });
+    } else {
+      updateOrder.mutate(
+        { id: localOrder.id, updatedOrder: createdOrder },
```
Comment (harrystead2) [inline src/components/CheckoutTabs/CheckoutSummaryTab.tsx]: I'm not sure we need the edit order here? We don't actually have an agreed idea of how a customer will go about editing their order - we need to think about that in the next milestone.

File: src/components/CheckoutFooter/CheckoutFooter.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+      <div className="flex w-full justify-between text-[11px] text-[#5A6675] transition-all duration-300">
+        <span>Service Charge:</span>
+        <span>{GBPFormat.format(SERVICE_CHARGE)}</span>
+      </div>
+
+      <div className="mt-2 mb-4 flex w-full justify-between text-[20px] font-[600] text-[#333F4C]">
+        <span>Total:</span>
+        <span>
+          {GBPFormat.format(localOrder.lineItems.reduce((partialSum, a) => partialSum + a.subTotal, 0) + (orderExists ? SERVICE_CHARGE : 0))}
```
Comment (harrystead2) [inline src/components/CheckoutFooter/CheckoutFooter.tsx]: could this be made into its own component, just to neaten things up, we use it at least 3 times elsewhere

File: src/components/CheckoutTabs/CheckoutConfirmationTab.tsx (hunk 2 above)
Comment (mbaltrimas) [reply src/components/CheckoutTabs/CheckoutConfirmationTab.tsx]: good point, added 👍

File: src/components/Inputs/InputRow.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+export const InputRow = ({ className, id, label, type, ...rest }: Props): React.ReactElement => {
+  return (
+    <div className={cn('flex w-full flex-col gap-1', className)}>
+      {label && (
+        <label htmlFor={id} className="font-poly italic text-[#A2AAB6]">
+          {label}
+        </label>
+      )}
+      <input type={type ?? 'text'} className="border-b-2 py-1" {...rest}></input>
```
Comment (harrystead2) [inline src/components/Inputs/InputRow.tsx]: need the input text to be #333F4C - 
image

File: src/context/CheckoutTabContext.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { createContext } from 'react';
+
```
Comment (harrystead2) [inline src/context/CheckoutTabContext.tsx:2]: Yeah this is good.

File: src/components/OrderDetails/OrderDetailsMinimal.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+                <span className="self-center">{lineItem.quantity}x </span>
+                <span className="self-center">{lineItem.product.name}</span>
+              </td>
+              <td>
+                <span className="text-center">{GBPFormat.format(lineItem.subTotal)}</span>
+              </td>
+            </tr>
+          ))
+        ) : (
+          <p>No items in order.</p>
```
Comment (harrystead2) [inline src/components/OrderDetails/OrderDetailsMinimal.tsx]: Don't need this, you're not allowed to progress from initial checkout page unless you have items in order

File: src/components/CheckoutTabs/CheckoutSummaryTab.tsx (hunk 3 above)
Comment (mbaltrimas) [reply src/components/CheckoutTabs/CheckoutSummaryTab.tsx]: i'll remove it for now

File: src/components/CheckoutFooter/CheckoutFooter.tsx (hunk 4 above)
Comment (mbaltrimas) [reply src/components/CheckoutFooter/CheckoutFooter.tsx]: donezo 👍

**Review by harrystead2 — DISMISSED** [review]

**Review by harrystead2 — APPROVED** [review]

**Review by bethcryer — APPROVED** [review]

**Pull request merged**